.github/workflows/validate-api.yaml

```diff
@@ -74,3 +79,113 @@ jobs:
         with:
           flags: api-e2e-tests
           files: /coverage/apps/api/coverage-final.json
+
+  validate-docker-build:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    name: Validate API (Docker)
+    needs: validate-node-build
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+
+      - name: Install pnpm
+        uses: pnpm/action-setup@v4
+        with:
+          version: 9.2.0
+          run_install: false
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Ensure Docker network
+        run: docker network inspect keyshade-ci >/dev/null 2>&1 || docker network create keyshade-ci
+
+      - name: Build API image
+        run: pnpm docker:build:api
+
+      - name: Verify API image
+        run: docker image inspect keyshade/api:ci-${{ github.sha }}
+
+      - name: Start API dependencies
+        run: |
+          set -euo pipefail
+
+          docker run -d --rm --name keyshade-ci-postgres --network keyshade-ci \
+            -e POSTGRES_USER=postgres \
+            -e POSTGRES_PASSWORD=postgres \
+            -e POSTGRES_DB=postgres \
+            postgres:15
+
+          docker run -d --rm --name keyshade-ci-redis --network keyshade-ci redis:7
+
+          for i in {1..30}; do
+            if docker exec keyshade-ci-postgres pg_isready -U postgres >/dev/null 2>&1; then
+              echo "Postgres is ready"
+              break
+            fi
+            if [ "$i" -eq 30 ]; then
+              echo "Postgres failed to become ready in time"
+              docker logs keyshade-ci-postgres || true
+              exit 1
+            fi
+            echo "Waiting for Postgres... ($i/30)"
+            sleep 2
+          done
+
+          for i in {1..30}; do
+            if docker exec keyshade-ci-redis redis-cli ping >/dev/null 2>&1; then
+              echo "Redis is ready"
+              break
+            fi
+            if [ "$i" -eq 30 ]; then
+              echo "Redis failed to become ready in time"
+              docker logs keyshade-ci-redis || true
+              exit 1
+            fi
+            echo "Waiting for Redis... ($i/30)"
+            sleep 2
+          done
+
+      - name: Run API container
+        run: pnpm docker:run:api
```

Comment:
Remove tthe envs that you are using in the `docker:run:api` command, and paste them here using the CI's environments. We want to keep the run command look lean